pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -435,6 +435,19 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 			status.Key(), status.DisplayName)
 		return false, true
 	}
+
+	verifyOnly := false
```

Comment:
Does this imply that when the controller tells the device to create a volume we will always defer the actual creation until there is a VolumeRefStatus with VerifyOnly=false?
That seems logical but just wanted to verify.

Do we need to do something special for baseos downloads?

Reply:
I want to keep existing logic for volumes that has no references from appInstances (and for baseOS ones as well, but good point, let me check), they will go through check here and will be created. But for volumes that are attached we need somehow indicate, that we want to defer the creation, and control this process from zedmanager via references.

Reply:
But since we have an inherent race where you can see a VolumeConfig directly from zedagent before there is any VolumeRefConfig from zedmanager I think you need to assume that if you do not see a VolumeRefConfig then verifyOnly needs to be assumed to be true?

Thus if there never is a volumerefconfig the volume would just progress to the loaded state.

Reply:
I also put check for `HasNoAppReferences` (from zedagent) here, this way I assume to be not affected by possible race.

Reply:
I forgot we  had HasNoAppReferences.

But it seems the logic is a bit backwards.

If we have a VolumeRefConfig, then we take verifyOnly from there.
Otherwise, if there is a VolumeConfig with HasNoAppReferences then verifyOnly=false (since the controller wants to create a volume independently of an application using it).
Or am I missing something?

Reply:
Correct, in case of HasNoAppReferences we will not change verifyOnly from false, I check here for !HasNoAppReferences, thus has references

Reply:
Modified